CMakeLists.txt

```
@@ -88,8 +88,7 @@ gz_find_package (Qt5
 
 #--------------------------------------
 # Find gz-sim
-gz_find_package(gz-sim10 REQUIRED PRIVATE COMPONENTS gui)
-set(GZ_SIM_VER ${gz-sim10_VERSION_MAJOR})
+gz_find_package(gz-sim REQUIRED PRIVATE COMPONENTS gui)
```

My personal opinion here is that a configuration instructions should encode the known versions that work and specially the versions that are known not to work.

If we know that this is going to fail with gz-sim less than 10, let's encode it.
```suggestion
gz_find_package(gz-sim REQUIRED VERSION 10 PRIVATE COMPONENTS gui)
```

---

I had suggested in the shorter proposal https://github.com/gazebo-tooling/release-tools/issues/1244#issuecomment-2730314344 that we not use explicit version numbers when calling `find_package` on the main branch, but that we add it in when creating stable release branches

> * Finding Gazebo packages with cmake's `find_package` function:
>    * On `main` branches: the major version number should not be used when finding a package with cmake’s `find_package()` function. For example: [find_package(gz-utils4 REQUIRED COMPONENTS log)](https://github.com/gazebosim/gz-utils/blob/7b4ac80941def8ced2c270bcd636f2fa4f24af1b/examples/log/CMakeLists.txt#L5) should instead be `find_package(gz-utils REQUIRED COMPONENTS log)`. This helps reduce maintenance churn on the `main` branches.
>    * When branching from `main` to create stable release branches: add explicit major versions into the `find_package` calls like `find_package(gz-utils 4 REQUIRED COMPONENTS log)`. This generates configuration errors when incompatible versions are built together, which is helpful.

I think this will reduce churn on the `main` branches

---

ah sorry, did not get that in mind. Sound fine.